diff -uNr a/configure.ac b/configure.ac
--- a/configure.ac	2017-07-26 22:39:07.461198820 +0100
+++ b/configure.ac	2017-07-26 22:39:53.284622741 +0100
@@ -31,7 +31,7 @@
 AC_CHECK_HEADERS([inttypes.h stdint.h malloc.h])
 AC_CHECK_FUNCS(setenv sinf)
 
-CXXFLAGS="$CXXFLAGS -fno-exceptions -Wall -ffast-math -fno-common -D_REENTRANT"
+CXXFLAGS="$CXXFLAGS -fno-exceptions -Wall -fno-common -D_REENTRANT"
 
 AC_CANONICAL_HOST
 case "$host" in
